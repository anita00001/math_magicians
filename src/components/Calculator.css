.calc-container {
  width: 30%;
}