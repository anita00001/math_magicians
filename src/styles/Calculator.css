.calc-wrapper {
  margin-top: 15vh;
  display: flex;
  justify-content: space-between;
}

.calc-wrapper p {
  font-size: 1.5rem;
  color: #5a4e4e;
  margin: 0;
  padding: 0;
  text-align: center;
  font-family: 'Pacifico', sans-serif;
  text-shadow: 1px 1px 1px black;
  text-decoration: underline overline;
  text-decoration-color: #5a4e4e;
  text-decoration-thickness: 0.1em;
  text-underline-offset: 0.2em;
  text-decoration-skip-ink: none;
}

.calc-container {
  width: 45%;
}

.output-field {
  width: 99%;
  height: 2.25rem;
  background-color: rgba(189, 184, 184, 0.9);
  border: none;
  text-align: right;
}
