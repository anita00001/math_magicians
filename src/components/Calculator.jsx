// Calculator.js
import React from 'react';
import Buttons from './Buttons';
import '../styles/Calculator.css';
import calculate from '../logic/calculate';

function Calculator() {
  const [output, setOutput] = React.useState('');
  const [calculatorData, setCalculatorData] = React.useState({});

  const handleButtonClick = (buttonLabel) => {
    const updatedData = calculate(calculatorData, buttonLabel);
    setCalculatorData(updatedData);

    if (buttonLabel === 'AC' || buttonLabel === '=') {
      setOutput(updatedData.total || '');
    } else if (buttonLabel === '+/-') {
      setOutput(updatedData.next || '');
    } else {
      setOutput((prevOutput) => prevOutput + buttonLabel);
    }
  };

  return (
    <div className="calc-wrapper">
      <div className="calc-container">
        <form>
          <input
            type="text"
            className="output-field"
            readOnly
            placeholder="0"
            value={output}
          />
        </form>
        <Buttons
          labels={[
            'AC', '+/-', '%', 'รท',
            '7', '8', '9', 'x',
            '4', '5', '6', '-',
            '1', '2', '3', '+',
            '0', '.', '=',
          ]}
          onButtonClick={handleButtonClick}
        />
      </div>
    </div>
  );
}

export default Calculator;
