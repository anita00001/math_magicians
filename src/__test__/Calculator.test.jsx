import { render, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator', () => {
  test('updates the output field correctly when buttons are clicked', () => {
    const { getByText, getByPlaceholderText } = render(<Calculator />);

    fireEvent.click(getByText('1'));
    fireEvent.click(getByText('+'));
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('='));

    const outputField = getByPlaceholderText('0');
    expect(outputField.value).toBe('3');
  });

  // snapshot test
  test('renders correctly', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });
});
