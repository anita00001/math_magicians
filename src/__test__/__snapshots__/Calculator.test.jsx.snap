// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<div>
  <div
    class="calc-wrapper"
  >
    <p>
      Let‘s do some Maths!
    </p>
    <div
      class="calc-container"
    >
      <form>
        <input
          class="output-field"
          placeholder="0"
          readonly=""
          type="text"
          value=""
        />
      </form>
      <div
        class="button-container"
      >
        <button
          class="calc-panel"
          type="button"
        >
          AC
        </button>
        <button
          class="calc-panel"
          type="button"
        >
          +/-
        </button>
        <button
          class="calc-panel"
          type="button"
        >
          %
        </button>
        <button
          class="oper-right"
          type="button"
        >
          ÷
        </button>
        <button
          class="calc-panel"
          type="button"
        >
          7
        </button>
        <button
          class="calc-panel"
          type="button"
        >
          8
        </button>
        <button
          class="calc-panel"
          type="button"
        >
          9
        </button>
        <button
          class="oper-right"
          type="button"
        >
          x
        </button>
        <button
          class="calc-panel"
          type="button"
        >
          4
        </button>
        <button
          class="calc-panel"
          type="button"
        >
          5
        </button>
        <button
          class="calc-panel"
          type="button"
        >
          6
        </button>
        <button
          class="oper-right"
          type="button"
        >
          -
        </button>
        <button
          class="calc-panel"
          type="button"
        >
          1
        </button>
        <button
          class="calc-panel"
          type="button"
        >
          2
        </button>
        <button
          class="calc-panel"
          type="button"
        >
          3
        </button>
        <button
          class="oper-right"
          type="button"
        >
          +
        </button>
        <button
          class="zero"
          type="button"
        >
          0
        </button>
        <button
          class="calc-panel"
          type="button"
        >
          .
        </button>
        <button
          class="oper-right"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
